moonraker/components/wled.py

```diff
@@ -245,31 +240,24 @@ def __init__(self: StripSerial,
         super().__init__(name, color_order, cfg)
 
         # Read the serial information (requires wled 0.13 2108250 or greater)
-        serialport: str = cfg.get("serial")
-        baud: int = cfg.getint("baud", 115200, above=49)
-
-        # write_timeout of 0 is non-blocking
-        self.ser = serial.Serial(serialport, baud,
-                                 write_timeout=0)
-        fd = self.ser.fileno()
-        os.set_blocking(fd, False)
+        self.serialport: str = cfg.get("serial")
+        self.baud: int = cfg.getint("baud", 115200, above=49)
 
     async def send_wled_command_impl(self: StripSerial,
                                      state: Dict[str, Any]) -> None:
         async with self.request_mutex:
-            logging.debug(f"WLED: serial:{self.ser.name} json:{state}")
+            if not hasattr(self, 'ser'):
+                _, self.ser = await serial_asyncio.open_serial_connection(
+                    url=self.serialport, baudrate=self.baud)
 
-            if not self.ser.is_open:
-                self.ser.open()
+            logging.debug(f"WLED: serial:{self.serialport} json:{state}")
 
-            # asyncio support is still experimental in pySerial
             self.ser.write(json.dumps(state).encode())
 
     def close(self: StripSerial):
         if self.ser.is_open:
```

Comment:
Presumably if the serial port fails to open `self.ser` may not be set, which would result in a an AttributeError here.  Also I'm not sure if `StreamWriter` has an `is_open` attribute.  You could probably call `is_closing()`.

Reply:
That's only set by close() so changed to a hasattr() check instead